config/src/config.rs

```diff
@@ -147,8 +147,40 @@ pub struct IndexerConfig {
 
 #[derive(Debug, Deserialize)]
 #[cfg_attr(test, derive(PartialEq))]
-pub struct DatabaseConfig {
-    pub postgres_url: Url,
+#[serde(untagged)]
+pub enum DatabaseConfig {
+    PostgresUrl {
+        postgres_url: Url,
+    },
+    PostgresVars {
+        host: String,
+        port: Option<u16>,
+        user: String,
+        password: Option<String>,
+        database: String,
+    },
+}
+impl DatabaseConfig {
+    pub fn get_formated_postgres_url(&self) -> Url {
+        match self {
+            DatabaseConfig::PostgresUrl { postgres_url } => postgres_url.clone(),
+            DatabaseConfig::PostgresVars {
```

Comment:
This is a little nit-picking but you can use just `self` and not clone. This is useful if you don't plan to use DatabaseConfig twice.